compiler/luci/pass/src/FuseBCQPass.cpp

```diff
@@ -93,32 +40,43 @@ template <> class BCQFuser<1>
 public:
   bool fuseBCQ(loco::Graph *g)
   {
-    for (auto node : loco::all_nodes(g))
+
+    const auto output_nodes = loco::output_nodes(g);
+    for (auto node : output_nodes)
     {
-      if (auto circle_const = dynamic_cast<luci::CircleConst *>(node))
+      auto output_node = loco::must_cast<luci::CircleOutput *>(node);
+      if ((int)output_node->index() > _original_output_cnt)
       {
-        add_BCQ_info_node(circle_const);
+        const auto prefix = (output_node->index() - (_original_output_cnt + 1)) / (_bundle_cnt);
+        const MetadataType metadata_type =
+            (MetadataType)((output_node->index() - (_original_output_cnt + 1)) % (_bundle_cnt));
```

Comment:
(optional) It would be better to use `static_cast` in c++. It does cast in compile time and make it more readable.